### ISSUE TITLE
autopull.sh of GNU Make git repo fails

### ISSUE THREAD

**Issue by komh** [bitwiseworks/wget-os2#2]
Hi/2.

If I try to launch `autopull.sh' of GNU Make git repo, it fails like:

```
./autopull.sh: getting translations into po/.reference for make...
Last-modified header missing -- time-stamps turned off.
2023-01-21 22:04:23 URL:https://translationproject.org/latest/make/ [6427/6427] -> "po/.reference/index.html.tmp" [1]
https://translationproject.org/robots.txt:
2023-01-21 22:04:23 ERROR 404: Not Found.
po/.reference/index.html?C=N;O=D.tmp: Bad file number
Cannot write to 'po/.reference/index.html?C=N;O=D.tmp' (Error 0).
po/.reference/index.html?C=M;O=A.tmp: Bad file number
Cannot write to 'po/.reference/index.html?C=M;O=A.tmp' (Error 0).
po/.reference/index.html?C=S;O=A.tmp: Bad file number
Cannot write to 'po/.reference/index.html?C=S;O=A.tmp' (Error 0).
po/.reference/index.html?C=D;O=A.tmp: Bad file number
Cannot write to 'po/.reference/index.html?C=D;O=A.tmp' (Error 0).
Last-modified header missing -- time-stamps turned off.
```

Or, you can use this command:

```
wget --mirror --level=1 -nd -nv -A.po -P "po/.reference" "https://translationproject.org/latest/make/"
```

This is because a local file name has '?' character.

Here is the patch.

[0001-Replace-with-3F-on-OS-2.patch](https://github.com/bitwiseworks/wget-os2/files/10472225/0001-Replace-with-3F-on-OS-2.patch)
```
From a604eaa4b226fe325bf7c6e067fa492f241c1d2d Mon Sep 17 00:00:00 2001
From: KO Myung-Hun <komh@chollian.net>
Date: Wed, 18 Jan 2023 21:55:50 +0900
Subject: [PATCH] Replace '?' with %3F on OS/2

'?', a separator of URL query string, is not allowed on OS/2.

* src/http.c (check_file_output) [OS/2]: Replace '?' with %3F.
---
 src/http.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/http.c b/src/http.c
index f61c99a..5c89125 100644
--- a/src/http.c
+++ b/src/http.c
@@ -2335,6 +2335,21 @@ check_file_output (const struct url *u, struct http_stat *hs,
       xfree (local_file);
     }
 
+#ifdef __OS2__
+    {
+      /* Convert '?' which is an invalid character on OS/2 to a percent
+       * encoding. */
+      char *p = strchr (hs->local_file, '?');
+      if (p)
+        {
+          char *tmp = aprintf ("%.*s%%3F%s",
+                               p - hs->local_file, hs->local_file, p + 1);
+          xfree (hs->local_file);
+          hs->local_file = tmp;
+        }
+    }
+#endif
+
   hs->temporary = opt.delete_after || opt.spider || !acceptable (hs->local_file);
   if (hs->temporary)
     {
-- 
2.30.0
```

Review, please...

**Comment by SilvanScherrer** [comment]
this seems ok. even I dislike the %3F in the file completely ;-) I will add it to the source soon, update to latest wget and release a new rpm

**Comment by SilvanScherrer** [comment]
fixed with 80c2f09b33f1e06afa17605abbf01cdbf1d4ecae
